inc/ticket.class.php

```diff
@@ -366,31 +366,44 @@ static function assignUserGroup(Ticket $ticket) {
             || $_SESSION['plugins']['escalade']['config']['use_assign_user_group_creation'] == 0) {
          return false;
       }
-
+      
       if (isset($ticket->input['_users_id_assign'])
           && $ticket->input['_users_id_assign'] > 0
           && (!isset($ticket->input['_groups_id_assign'])
             || $ticket->input['_groups_id_assign'] <= 0)) {
-         if ($_SESSION['plugins']['escalade']['config']['use_assign_user_group'] == 1) {
-            // First group
-            $ticket->input['_groups_id_assign']
-               = PluginEscaladeUser::getTechnicianGroup($ticket->input['entities_id'],
-                                                         $ticket->input['_users_id_assign'],
-                                                         true);
-            //prevent adding empty group
-            if (empty($ticket->input['_groups_id_assign'])) {
-               unset($ticket->input['_groups_id_assign']);
-            }
-         } else {
-            // All groups
-            $ticket->input['_additional_groups_assigns']
-               = PluginEscaladeUser::getTechnicianGroup($ticket->input['entities_id'],
-                                                         $ticket->input['_users_id_assign'],
-                                                         false);
-            //prevent adding empty group
-            if (empty($ticket->input['_additional_groups_assigns'])) {
-               unset($ticket->input['_additional_groups_assigns']);
-            }
+         switch ($_SESSION['plugins']['escalade']['config']['use_assign_user_group']) {
+            case 1 : 
+               // First group
+               $ticket->input['_groups_id_assign']
+                  = PluginEscaladeUser::getTechnicianGroup($ticket->input['entities_id'],
+                                                            $ticket->input['_users_id_assign'],
+                                                            true);
+               //prevent adding empty group
+               if (empty($ticket->input['_groups_id_assign'])) {
+                  unset($ticket->input['_groups_id_assign']);
+               }
+               break;
+            case 2:
+               // All groups
+               $ticket->input['_additional_groups_assigns']
+                  = PluginEscaladeUser::getTechnicianGroup($ticket->input['entities_id'],
+                                                            $ticket->input['_users_id_assign'],
+                                                            false);
+               //prevent adding empty group
+               if (empty($ticket->input['_additional_groups_assigns'])) {
+                  unset($ticket->input['_additional_groups_assigns']);
+               }
+               break;
+            
+            case 3:
+               $groups_id = PluginEscaladeUser::getUserDefaultGroup($item->fields['users_id']);
```

Comment:
I guess returned value should be added to either `$ticket->input['_additional_groups_assigns']` or `$ticket->input['_groups_id_assign']`.

By the way, I do not understand why value is affected to different input key for cases 1 and 2.

Reply:
I can't tell you more. More code comments would be better in order to understand why and how this part of the code is used.
I've fixed the method return variable